cmd/auth/main.go

```diff
@@ -61,20 +63,23 @@ const (
-	JaegerURL           url.URL       `env:"SMQ_JAEGER_URL"                   envDefault:"http://localhost:4318/v1/traces"`
-	SendTelemetry       bool          `env:"SMQ_SEND_TELEMETRY"               envDefault:"true"`
-	InstanceID          string        `env:"SMQ_AUTH_ADAPTER_INSTANCE_ID"     envDefault:""`
-	AccessDuration      time.Duration `env:"SMQ_AUTH_ACCESS_TOKEN_DURATION"   envDefault:"1h"`
-	RefreshDuration     time.Duration `env:"SMQ_AUTH_REFRESH_TOKEN_DURATION"  envDefault:"24h"`
-	InvitationDuration  time.Duration `env:"SMQ_AUTH_INVITATION_DURATION"     envDefault:"168h"`
-	SpicedbHost         string        `env:"SMQ_SPICEDB_HOST"                 envDefault:"localhost"`
-	SpicedbPort         string        `env:"SMQ_SPICEDB_PORT"                 envDefault:"50051"`
-	SpicedbSchemaFile   string        `env:"SMQ_SPICEDB_SCHEMA_FILE"          envDefault:"./docker/spicedb/schema.zed"`
-	SpicedbPreSharedKey string        `env:"SMQ_SPICEDB_PRE_SHARED_KEY"       envDefault:"12345678"`
-	TraceRatio          float64       `env:"SMQ_JAEGER_TRACE_RATIO"           envDefault:"1.0"`
-	ESURL               string        `env:"SMQ_ES_URL"                       envDefault:"nats://localhost:4222"`
+	LogLevel                   string        `env:"SMQ_AUTH_LOG_LEVEL"                envDefault:"info"`
+	SecretKey                  string        `env:"SMQ_AUTH_SECRET_KEY"               envDefault:"secret"`
+	JaegerURL                  url.URL       `env:"SMQ_JAEGER_URL"                    envDefault:"http://localhost:4318/v1/traces"`
+	SendTelemetry              bool          `env:"SMQ_SEND_TELEMETRY"                envDefault:"true"`
+	InstanceID                 string        `env:"SMQ_AUTH_ADAPTER_INSTANCE_ID"      envDefault:""`
+	AccessDuration             time.Duration `env:"SMQ_AUTH_ACCESS_TOKEN_DURATION"    envDefault:"1h"`
+	RefreshDuration            time.Duration `env:"SMQ_AUTH_REFRESH_TOKEN_DURATION"   envDefault:"24h"`
+	InvitationDuration         time.Duration `env:"SMQ_AUTH_INVITATION_DURATION"      envDefault:"168h"`
+	SpicedbHost                string        `env:"SMQ_SPICEDB_HOST"                  envDefault:"localhost"`
+	SpicedbPort                string        `env:"SMQ_SPICEDB_PORT"                  envDefault:"50051"`
+	SpicedbSchemaFile          string        `env:"SMQ_SPICEDB_SCHEMA_FILE"           envDefault:"./docker/spicedb/schema.zed"`
+	SpicedbPreSharedKey        string        `env:"SMQ_SPICEDB_PRE_SHARED_KEY"        envDefault:"12345678"`
+	TraceRatio                 float64       `env:"SMQ_JAEGER_TRACE_RATIO"            envDefault:"1.0"`
+	ESURL                      string        `env:"SMQ_ES_URL"                        envDefault:"nats://localhost:4222"`
+	AuthCalloutURLs            []string      `env:"SMQ_AUTH_CALLOUT_URLS"             envDefault:"" envSeparator:","`
+	AuthCalloutMethod          string        `env:"SMQ_AUTH_CALLOUT_METHOD"           envDefault:"POST"`
```

Comment:
we can try them iteratively on failure POST, then GET

Reply:
I prefer not to add these mechanisms since we already have the flexibility to specify the HTTP method type in the environment configuration. This allows us to manage different request methods effectively without adding extra logic for retries. If the user mistakenly provides POST instead of GET, it would be their responsibility to correct this in the configuration.